tools/convert_tests/convert_tests_helper.py

```diff
@@ -0,0 +1,409 @@
+        pass  # Continue to the next format if parsing fails
+
+    # Fallback to parsing without fractional seconds
+    try:
+        dt = datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+        return dt.isoformat()
+    except ValueError:
+        # Return the original value if it doesn't match expected formats
+        return value
+
+
+def has_last_colon_after_last_dash(s):
+    """
+    Check if the last ':' comes after the last '-' in the string.
+
+    Args:
+        s (str): Input string, e.g., "-10:30".
+
+    Returns:
+        bool: True if the last ':' comes after the last '-', False otherwise.
+    """
+    last_dash_index = s.rfind("-")  # Find the last occurrence of '-'
+    last_colon_index = s.rfind(":")  # Find the last occurrence of ':'
+
+    # Check if both '-' and ':' exist and if last ':' comes after last '-'
+    return (
+        last_dash_index != -1
+        and last_colon_index != -1
+        and last_dash_index < last_colon_index
+    )
+
+
+def format_timestamp_tz(timestamp_with_tz):
+    """Convert a timestamp with timezone abbreviation to ISO 8601 with offset."""
+    if "-" in timestamp_with_tz and not has_last_colon_after_last_dash(
+        timestamp_with_tz
+    ):
+        return timestamp_with_tz.replace(" ", "T") + ":00"
+    datetime_str, tz_abbr = timestamp_with_tz.rsplit(" ", 1)
+    dt = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+
+    if tz_abbr in timezone_abbr_map:
+        dt = pytz.timezone(timezone_abbr_map[tz_abbr]).localize(dt)
+    else:
+        return timestamp_with_tz
+        # raise ValueError(f"Invalid timezone abbreviation: {tz_abbr}")
+
+    return dt.isoformat()
+
+
+def convert_sql_interval_to_iso_duration(interval: str):
+    """Convert SQL-style interval to ISO 8601 format."""
+    # Regular expression to match interval components (e.g., '1 DAY 10 HOUR' or '360 DAY, 1 HOUR')
+    match = re.match(
+        r"INTERVAL\s*'((\d+)\s*(YEAR|MONTH|DAY|HOUR|MINUTE|SECOND)(\s+\d+\s*(YEAR|MONTH|DAY|HOUR|MINUTE|SECOND))*)'",
+        interval.strip(),
+    )
+    if not match:
+        raise ValueError(f"Invalid SQL interval format: {interval}")
+
+    # Capture all parts of the interval, including multiple parts (space-separated)
+    parts = match.group(1).split()
+
+    iso_parts = []
+    unit_map = {
+        "YEAR": "P{}Y",
+        "MONTH": "P{}M",
+        "DAY": "P{}D",
+        "HOUR": "PT{}H",
+        "MINUTE": "PT{}M",
+        "SECOND": "PT{}S",
+    }
+
+    i = 0
+    iyear = False
+    while i < len(parts):
+        num = int(parts[i])  # The number part
+        unit = parts[i + 1]  # The unit part
+        if unit in ["YEAR", "MONTH"]:
+            iyear = True
+        iso_parts.append(unit_map[unit].format(num))
+        i += 2  # Skip to the next number-unit pair
+
+    # Join the components together to form the final ISO 8601 duration string
+    iso_interval = "".join(iso_parts)
+
+    # Determine whether it's a 'iyear' or 'iday' based on units
+    if iyear:
```

Comment:
This won't hold for compound intervals, but we won't have those in YAML tests.  Add a comment to state that we only will see these two types?